src/main/java/org/tikv/raw/RawKVClient.java

```diff
@@ -1,20 +1,38 @@
+/*
+ * Copyright 2018 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package org.tikv.raw;
 
 import com.google.protobuf.ByteString;
-import java.util.*;
 import org.tikv.common.TiConfiguration;
 import org.tikv.common.TiSession;
+import org.tikv.common.exception.TiKVException;
 import org.tikv.common.operation.iterator.RawScanIterator;
 import org.tikv.common.region.RegionManager;
 import org.tikv.common.region.RegionStoreClient;
 import org.tikv.common.region.TiRegion;
+import org.tikv.common.util.BackOffFunction;
 import org.tikv.common.util.BackOffer;
 import org.tikv.common.util.ConcreteBackOffer;
 import org.tikv.common.util.Pair;
 import org.tikv.kvproto.Kvrpcpb;
 import org.tikv.kvproto.Metapb;
 
-public class RawKVClient {
+import java.util.*;
+
+public class RawKVClient implements AutoCloseable {
```

Comment:
Why not implement `Closeable` too?

Reply:
Shouldn't do both. Closeable extends AutoCluseable with more specific exception type (IOException). Closeable is kind of legacy interface for old version of JDK. Most of the exception thrown from inside session is not IOException, more specifically,  we here prefer unchecked exception.

Reply:
Sounds good.